src/NServiceBus.AmazonSQS/Configure/TransportInfrastructure.cs

```diff
@@ -20,8 +21,27 @@ public TransportInfrastructure(ReadOnlySettings settings)
         {
             configuration = new TransportConfiguration(settings);
 
-            sqsClient = configuration.SqsClientFactory();
-            s3Client = configuration.S3ClientFactory();
+            try
+            {
+                sqsClient = configuration.SqsClientFactory();
+            }
+            catch (Exception e) // ideally this would just raise AmazonClientException but the fullframework and .net core version behave different, see https://github.com/aws/aws-sdk-net/issues/1216
```

Comment:
@mauroservienti I had to update the PR. @kbaley and I found a different behavior between the full framework and .NET Core version of it, more context available in https://github.com/aws/aws-sdk-net/issues/1216

Reply:
Wouldn’t it be better to catch different exceptions using compiler directives? By catching the broader Exception can’t we run the risk of hiding different kind of errors under the missing credentials umbrella?

Reply:
If we did that then we would have to include a package reference in the .NET core specific part to System.Net.Http.

I know that this catch is broad but any exception when creating the client hints either at credential problems or like the generic phrase

> or the client factory configures the created client accordingly

hints a broader problem with client configuration. The inner exception is still logged so while still have the generic catch the log would contain more information to troubleshoot in case we'd run into a more broader problem and the endresult would still be that the endpoint cannot start.